### PR TITLE
validate zip file entries prior to decompression

### PR DESCRIPTION
**Pull request by mikehardy**
zip file path traversal was a potential problem in the code

reference discussion [here](https://wiki.sei.cmu.edu/confluence/display/java/IDS04-J.+Safely+extract+files+from+ZipInputStream)

Pointed out by Danny Grander of Snyk Security Research Team

The reference discussion also handles disk space issues, which we could do better on, but I wanted to keep this proposed change tiny.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java [hunk 1]
```diff
@@ -627,9 +627,12 @@ public static void unzipFiles(ZipFile zipFile, String targetDirectory, String[]
                 ZipEntry ze = zipFile.getEntry(requestedEntry);
                 if (ze != null) {
                     String name = ze.getName();
+
```
[the remark this answers is not in the document]
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java]: sorry about that

File: AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java [hunk 2]
```diff
@@ -655,6 +658,19 @@ public static void unzipFiles(ZipFile zipFile, String targetDirectory, String[]
         }
     }
 
+    private static String validateFilename(String filename, File iD)
```
[the remark this answers is not in the document]
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java]: agreed, that is much cleaner, pushing changes in a moment

**Comment by mikehardy** [comment]
Changes squashed+pushed and CI passed, I was so focused on implementing the functionality I clearly didn't think about the signature- a general problem with programming rustiness I think. Thanks for the review

**Comment by timrae** [comment]
Thanks for taking care of this!

**Pull request merged**